spring-cloud-kubernetes-controllers/spring-cloud-kubernetes-discoveryserver/src/main/java/org/springframework/cloud/kubernetes/discoveryserver/DiscoveryServerController.java

```diff
@@ -56,8 +58,4 @@ public Mono<ServiceInstance> appInstance(@PathVariable String name, @PathVariabl
 				.filter(serviceInstance -> serviceInstance.getInstanceId().equals(instanceId)).singleOrEmpty();
 	}
 
-	record Service(String name, List<ServiceInstance> serviceInstances) {
```

Comment:
1. the idea of this PR really starts from here. So in discovery server we have a "Service", since we talk over http with the client, at the time of the creation it meant to create a "Service" in the client also. But since then, we have refactored the code a lot, and now its possible to move this logical piece of Service into the commons package. This is what I am doing here: remove from here, move to commons